processing/data_collection/gazette/spiders/al_maceio.py

```diff
@@ -0,0 +1,74 @@
+from dateparser import parse
+from datetime import datetime
+
+import scrapy
+
+from gazette.items import Gazette
+
+
+class AlMaceioSpider(scrapy.Spider):
+    MUNICIPALITY_ID = '2704302'
+    name = 'al_maceio'
+    allowed_domains = ['maceio.al.gov.br']
+    start_urls = ['http://www.maceio.al.gov.br/noticias/diario-oficial/']
+    date_limit = datetime(2015, 1, 1)
+    page_number = 1
+
+    def parse(self, response):
+        """
+        @url http://www.maceio.al.gov.br/noticias/diario-oficial/
+        @returns items 0 9
+        @returns requests 1 10
+        @scrapes date file_urls is_extra_edition municipality_id power scraped_at
+        """
+        for gazette in response.xpath('//article'):
+            url = gazette.xpath('a/@href').extract_first()
+            date_str = gazette.xpath('time/text()').extract_first()
+            date = parse(date_str, languages=['pt'])
+            title = gazette.xpath('a/@title').extract_first()
+            is_extra_edition = 'suplemento' in (title.lower())
+
+            if date >= self.date_limit:
+                if 'wp-content/uploads' in url:
+                    gazette = self.create_gazette(date, url, is_extra_edition)
+                    yield gazette
+                else:
+                    request = scrapy.Request(url, self.parse_additional_page)
+                    request.meta['date'] = date
+                    request.meta['is_extra_edition'] = is_extra_edition
+                    yield request
+            else:
+                break
+
+        """
+        This condition is necessary to stop crawling when date < date_limit.
+        """
+        if date >= self.date_limit:
```

Comment:
See https://github.com/okfn-brasil/diario-oficial/pull/41/ and https://github.com/okfn-brasil/diario-oficial/pull/46 .
There is no need to include date validation in your spider. This will be handled in an item pipeline.